rules/onchain_budget_test.go

```diff
@@ -0,0 +1,281 @@
+package rules
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/stretchr/testify/require"
+)
+
+// TestOnChainBudgetVerifySane tests that the OnChainBudgetMgr VerifySane method
+// correctly verifies the value of the rate limit depending on given min sane
+// value.
+func TestOnChainBudgetVerifySane(t *testing.T) {
+	var min = &OnChainBudget{
+		AbsoluteAmtSats: 1000,
+		MaxSatPerVByte:  30,
+	}
+
+	tests := []struct {
+		name      string
+		values    *OnChainBudget
+		expectErr bool
+	}{
+		{
+			name: "within bounds",
+			values: &OnChainBudget{
+				AbsoluteAmtSats: 2000,
+				MaxSatPerVByte:  40,
+			},
```

Comment:
A general comment for the bounds tests: IMO when we're testing bounds, it makes sense to also have a test that sets these values at the exact bounds i.e. `1000` & `30`. Then the tests, like "amount too low" below should lower the amount by the least possible value (i.e. `999`) and keep the `MaxSatPerVByte` at the exact limit.

That way the unit tests will be much more precise and catch potential bugs where the code uses a `<=` comparison instead of a `<` etc.

Hope that makes sense?

I do think that it does make sense to keep the current "within bounds" test though, just to prove that the values don't need to be at the exact bounds :)

Reply:
Makes sense :+1:!